### PR TITLE
Graph App Installation Life Cycle using Tab csharp

### PR DESCRIPTION
**Pull request by Trinetra-MSFT**
## Description
This sample app demonstrates the installation life cycle of teams app using Graph API.

Language Used: C#

Following scenario are covered in configurable tab for app installation life cycle
1. Covered List apps
2. Update app
3. Delete app
4. Upgrade app
5. Add app for user
Following permissions needs to be added at Application level .
image

### CODE REVIEW

File: samples/graph-app-installation-lifecycle/csharp/AppInstallation/Controllers/HomeController.cs [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+
+        [HttpPost]
+        [Route("AddApp")]
+        public async Task<string> AddApp(string tenantId, string teamId)
+        {
+            string token = await GetToken(tenantId);
+            GraphServiceClient graphClient = GetAuthenticatedClient(token);
+
+            var appsInstalled = await graphClient.Teams[teamId].InstalledApps.Request().Expand("teamsAppDefinition").GetAsync();
+            var teamsAppId = appsInstalled.Where(o => o.TeamsAppDefinition.DisplayName == "App Installation").Select(o => o.TeamsAppDefinition.TeamsAppId).FirstOrDefault();
```
Comment (Wajeed-msft) [inline samples/graph-app-installation-lifecycle/csharp/AppInstallation/Controllers/HomeController.cs:112]: Does installing same app works?

File: samples/graph-app-installation-lifecycle/csharp/AppInstallation/Views/Home/Index.cshtml [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+<script type="text/javascript">
+    $(document).ready(function () {
+        microsoftTeams.initialize();
+        var teamAppInstallationId = '';
+
+            $(".btngetApp").on('click', function (e) {
+                microsoftTeams.getContext(function (context) {
+                    var tenantId = context.tid;
+                    var teamId = context.groupId;
```
Comment (Wajeed-msft) [inline samples/graph-app-installation-lifecycle/csharp/AppInstallation/Views/Home/Index.cshtml:93]: We are fetching context evetime, instead you can call it once and save the information in variable and reuse.

File: samples/graph-app-installation-lifecycle/csharp/AppInstallation/AppInstallation.csproj [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿<Project Sdk="Microsoft.NET.Sdk.Web">
```
Comment (Wajeed-msft) [inline samples/graph-app-installation-lifecycle/csharp/AppInstallation/AppInstallation.csproj:1]: Could you please add entry in main read me file under Graph?

File: samples/graph-app-installation-lifecycle/csharp/AppInstallation/Manifest/manifest.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
+{
```
Comment (Wajeed-msft) [inline samples/graph-app-installation-lifecycle/csharp/AppInstallation/Manifest/manifest.json:1]: Where are we using images present in this folder?  https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/v-trkum/appInstallation/samples/graph-app-installation-lifecycle/csharp/AppInstallation/Images

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/graph-app-installation-lifecycle/csharp/AppInstallation/Views/Home/Index.cshtml (hunk 2 above)
Comment (Trinetra-MSFT) [reply samples/graph-app-installation-lifecycle/csharp/AppInstallation/Views/Home/Index.cshtml]: Updated this comment

File: samples/graph-app-installation-lifecycle/csharp/AppInstallation/AppInstallation.csproj (hunk 3 above)
Comment (Trinetra-MSFT) [reply samples/graph-app-installation-lifecycle/csharp/AppInstallation/AppInstallation.csproj:1]: added link in Root ReadMe file also update app readMe

File: samples/graph-app-installation-lifecycle/csharp/AppInstallation/Manifest/manifest.json (hunk 4 above)
Comment (Trinetra-MSFT) [reply samples/graph-app-installation-lifecycle/csharp/AppInstallation/Manifest/manifest.json:1]: done

File: samples/graph-app-installation-lifecycle/csharp/AppInstallation/Controllers/HomeController.cs (hunk 1 above)
Comment (Trinetra-MSFT) [reply samples/graph-app-installation-lifecycle/csharp/AppInstallation/Controllers/HomeController.cs]: Yes, it works fine, Update this comment

File: samples/graph-app-installation-lifecycle/csharp/README.md [hunk 5]
```diff
@@ -18,6 +18,8 @@ This sample app demonstarte the installation lifecycle for Teams [Apps](https://
 
 - Register your app with Microsoft identity platform via the Azure AD portal
   - Your app must be registered in the Azure AD portal to integrate with the Microsoft identity platform and call Microsoft Graph APIs. See [Register an application with the Microsoft identity platform](https://docs.microsoft.com/en-us/graph/auth-register-app-v2). 
+- You need to add following permissions mentioned in the below screenshots to call respective Graph   API
+![](https://user-images.githubusercontent.com/50989436/115846799-e2ce5500-a43f-11eb-984a-b16aec0bc586.png))
```
Comment (Wajeed-msft) [inline samples/graph-app-installation-lifecycle/csharp/README.md:22]: These are not required. Please check and update.

**Review by Wajeed-msft — CHANGES_REQUESTED** [review]

File: samples/graph-app-installation-lifecycle/csharp/README.md (hunk 5 above)
Comment (Trinetra-MSFT) [reply samples/graph-app-installation-lifecycle/csharp/README.md]: updated

**Review by Wajeed-msft — APPROVED** [review]

**Review by asnow003 — APPROVED** [review]

**Pull request merged**